tests/proxy.spec.ts

```diff
@@ -75,6 +76,62 @@ it('should work with IP:PORT notion', async ({browserType, browserOptions, serve
   await browser.close();
 });
 
+it('should not work with localhost by default on Chromium (unless Linux Headful)', async ({browserType, platform, browserOptions }) => {
+  let proxiedRequest = false;
+  const server = net.createServer(socket => {
+    socket.on('data', () => {
+      proxiedRequest = true;
+      socket.end();
+    });
+  });
+  await new Promise(f => server.listen(0, f));
+
+  const opts: LaunchOptions = {
+    ...browserOptions,
+    proxy: { server: `http://127.0.0.1:${(server.address() as any).port}` },
+  };
+
+  const browser = await browserType.launch(opts);
+
+  const page = await browser.newPage();
+
+  await page.goto('http://localhost:9999/request').catch(() => {});
+  await browser.close();
+  server.close();
+  if (browserType.name() !== 'chromium' || platform === 'linux' && !browserOptions.headless)
```

Comment:
From previous CI run, it turns out Headful Chrome on Linux is unique and does pass by default: https://github.com/microsoft/playwright/runs/3356777516?check_suite_focus=true